Task1/Project/Shapes/Models/Classes/ShapeBase.cs

```diff
@@ -50,89 +76,89 @@ public CoordinateQuarters GetQuarter
         /// </returns>
         protected abstract Point GetMiddlePoint();
         /// <summary>
-        /// When overridden in a derived class, interprets string as numeric data.
+        /// Register shapes in dictionary of Id.
         /// </summary>
-        /// <param name="line">
-        /// The string data.
+        /// <param name="key">
+        /// Id of the shape.
         /// </param>
-        protected abstract void Interpret(string line);       
+        /// <param name="transformer">
+        /// Transformer <see cref="string"/> to <see cref="ShapeBase"/>.
+        /// </param>
+        public static void RegisterShape(string key, Deserialization transformer)

```

Comment:
should take a **System.Type** to registrate as a first argument

Reply:
and it is not a transformer, its probably more as creator, or deserializator

Reply:
I wrote "creator". It will be normal, I think.

Reply:
it shall be